packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/List/PreviewItem.tsx

```diff
@@ -0,0 +1,221 @@
+import {
+  Box,
+  Button,
+  Card,
+  CardTone,
+  Flex,
+  Menu,
+  MenuButton,
+  MenuItem,
+  Spinner,
+  Text,
+} from '@sanity/ui'
+import React, {useCallback, useMemo, useRef} from 'react'
+import {SchemaType} from '@sanity/types'
+import {CopyIcon as DuplicateIcon, EllipsisVerticalIcon, TrashIcon} from '@sanity/icons'
+import {getSchemaTypeTitle} from '../../../../../schema'
+import {ObjectItem, ObjectItemProps} from '../../../../types'
+import {useScrollIntoViewOnFocusWithin} from '../../../../hooks/useScrollIntoViewOnFocusWithin'
+import {useDidUpdate} from '../../../../hooks/useDidUpdate'
+import {useChildPresence} from '../../../../studio/contexts/Presence'
+import {randomKey} from '../../../../utils/randomKey'
+import {FormFieldValidationStatus} from '../../../../components'
+import {FieldPresence} from '../../../../../presence'
+import {useChildValidation} from '../../../../studio/contexts/Validation'
+import {ChangeIndicator} from '../../../../../changeIndicators'
+import {RowLayout} from '../../layouts/RowLayout'
+import {createProtoArrayValue} from '../createProtoArrayValue'
+import {InsertMenu} from '../InsertMenu'
+import {EditPortal} from '../../../../components/EditPortal'
+
+type PreviewItemProps<Item extends ObjectItem> = Omit<ObjectItemProps<Item>, 'renderDefault'>
+
+function getTone({
+  readOnly,
+  hasErrors,
+  hasWarnings,
+}: {
+  readOnly: boolean | undefined
+  hasErrors: boolean
+  hasWarnings: boolean
+}): CardTone {
+  if (readOnly) {
+    return 'transparent'
+  }
+  if (hasErrors) {
+    return 'critical'
+  }
+  return hasWarnings ? 'caution' : 'default'
+}
+const MENU_POPOVER_PROPS = {portal: true, tone: 'default'} as const
+
+export function PreviewItem<Item extends ObjectItem = ObjectItem>(props: PreviewItemProps<Item>) {
+  const {
+    schemaType,
+    parentSchemaType,
+    path,
+    readOnly,
+    onRemove,
+    value,
+    open,
+    onInsert,
+    onFocus,
+    onOpen,
+    onClose,
+    changed,
+    focused,
+    children,
+    inputProps: {renderPreview},
+  } = props
+
+  const sortable = parentSchemaType.options?.sortable !== false
+  const insertableTypes = parentSchemaType.of
+
+  const previewCardRef = useRef<HTMLDivElement | null>(null)
+
+  // this is here to make sure the item is visible if it's being edited behind a modal
+  useScrollIntoViewOnFocusWithin(previewCardRef, open)
+
+  useDidUpdate(focused, (hadFocus, hasFocus) => {
+    if (!hadFocus && hasFocus && previewCardRef.current) {
+      // Note: if editing an inline item, focus is handled by the item input itself and no ref is being set
+      previewCardRef.current?.focus()
+    }
+  })
+
+  const resolvingInitialValue = (value as any)._resolvingInitialValue
+
+  const handleDuplicate = useCallback(() => {
```

Comment:
In the same vein as for `useArrayHandlers`suggestion: `useArrayItemHandlers(props: PreviewItemProps)`which returns {handleDuplicate, handleInsert} (other things?) and share with GridItem